anjali/sb-pt-cont/3.js

```diff
@@ -0,0 +1,105 @@
+// https://levelbuilder-studio.code.org/s/sb-pt-cont/lessons/1/levels/3
+/*
+Make two sprites with different costumes and different locations. Use the pin on the location block to move sprites to different locations.
+*/
+if (!validationProps.successCriteria) {
+  validationProps.successCriteria = {
+    twoSprites: false,
+    differentLocations: false,
+    differentCostumes: false
+  };
+}
+
+//Helper variables
+var spriteIds = getSpriteIdsInUse();
+
+//Count Sprites
+validationProps.successCriteria.twoSprites = spriteIds.length>=2;
+
+if (World.frameCount == 1) {
+  // Check sprite locations
+  var uniqueStartingSpriteLocations = [];
```

Comment:
Let's try to do this with a set rather than an array (both for clarity and performance).
You should be able to do something sort of like this:
```
var x = new Set()
var c1 = {x: 200, y: 200}
var c2 = {x: 150, y: 200}

x.add("(" + c1.x + "," + c1.y + ")")
x.add("(" + c2.x + "," + c2.y + ")")
x.add("(" + c1.x + "," + c1.y + ")")

x
> Set(2) {"(200,200)", "(150,200)"}
```
Then you can just check if the length of the set is the same as the number of sprites

Reply:
I originally did this with a Set as you suggested, but I get the error `ReferenceError: Set is not defined`. Do I have to import Set in JS? I didn't think so...